src/compiler/checker.ts

```diff
@@ -34681,7 +34679,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 right,
                 Diagnostics.Property_0_is_not_accessible_outside_class_1_because_it_has_a_private_identifier,
                 diagName,
-                diagnosticName(typeClass.name || anon),

```

Comment:
I think `anon` can now be completely removed. Only `checkNoTypeArguments` relies on it - but I can't see how it would actually end up in that branch (and there are no tests proving otherwise). I have refrained myself from doing changes there though.